wrapanapi/nuage.py

```diff
@@ -53,26 +65,206 @@ def common_headers(self):
             'X-Nuage-Organization': 'csp'
         }
 
+    #
+    # Integration-test specific entities on Nuage server
+    #
+
+    @property
+    def enterprise(self):
+        if not self._enterprise:
+            match = self.list_enterprises(name=self.enterprise_name)
+            if match:
+                self._enterprise = match[0]['ID']
+        return self._enterprise
+
+    @property
+    def domain_template(self):
+        if not self._domain_template:
+            match = self.list_domain_templates(name=self.domain_template_name)
+            if match:
+                self._domain_template = match[0]['ID']
+        return self._domain_template
+
+    @property
+    def domain(self):
+        if not self._domain:
+            match = self.list_domains_for_enterprise(name=self.domain_name)
+            if match:
+                self._domain = match[0]['ID']
+        return self._domain
+
+    @property
+    def zone(self):
+        if not self._zone:
+            match = self.list_zones_for_domain(name=self.zone_name)
+            if match:
+                self._zone = match[0]['ID']
+        return self._zone
+
+    @property
+    def subnet(self):
+        if not self._subnet:
+            match = self.list_cloud_subnets(name=self.subnet_name)
+            if match:
+                self._subnet = match[0]['ID']
+        return self._subnet
+
+    #
+    # Multi API calls wrappers
+    #
+
+    def destroy_enterprise(self, name=None):
+        """Delete enterprise by name if exists, even if it's non-empty"""
+        if not name:
+            name = self.enterprise_name
+        enterprise = [etp['ID'] for etp in self.list_enterprises() if etp['name'] == name]
+        if not enterprise:  # enterprise doesn't exist so we are done
+            return
+        else:
+            enterprise = enterprise[0]
+        for domain in self.list_domains_for_enterprise(enterprise):
+            self.delete_domain(domain['ID'])
+        self.delete_enterprise(enterprise)
+
+    #
+    # API calls
+    #
+
     def list_network_groups(self):
         return self._request_list('/enterprises', 'get')
 
-    def list_cloud_subnets(self):
-        return self._request_list('/subnets', 'get', exclude_name='BackHaulSubnet')
+    def list_cloud_subnets(self, name=None):
+        return self._request_list(
+            '/subnets',
+            'get',
+            exclude_name='BackHaulSubnet',
+            search_name=name
+        ) or []
+
+    def list_cloud_subnets_for_domain(self, domain=None):
+        if not domain:
```

Comment:
`domain = domain or self.domain` - can probably collapse that into the statement below even, though that might not format well within 100 chars.